src/Orleans/Core/Grain.cs

```diff
@@ -332,5 +324,51 @@ protected virtual Task ReadStateAsync()
         {
             return storage.ReadStateAsync();
         }
+
+        public void Participate(ILifecycleObservable<GrainLifecycleStage> lifecycle)
+        {
+            lifecycle.Subscribe(GrainLifecycleStage.SetupState, this);
+        }
+
+        public async Task OnStart(CancellationToken ct)
```

Comment:
suggest we make this an explicit implementation so as not to pollute intellisense for Grain